code/FulpstationCode/sec_borg_files/fulp_sec_borg.dm

```diff
@@ -0,0 +1,477 @@
+	if(iscyborg(user))
+		var/mob/living/silicon/robot/R = user
+
+		if(shock)
+			to_chat(user, "<span class='notice'>You clear all active holograms, and reset your projector to normal.</span>")
+			holosign_type = /obj/structure/holosign/barrier
+			creation_time = 5
+			if(signs.len)
+				for(var/H in signs)
+					qdel(H)
+			shock = 0
+			return
+		else if(R.emagged&&!shock)
+			to_chat(user, "<span class='warning'>You clear all active holograms, and overload your energy projector!</span>")
+			holosign_type = /obj/structure/holosign/barrier/cyborg/hacked
+			creation_time = 30
+			if(signs.len)
+				for(var/H in signs)
+					qdel(H)
+			shock = 1
+			return
+		else
+			if(signs.len)
+				for(var/H in signs)
+					qdel(H)
+				to_chat(user, "<span class='notice'>You clear all active holograms.</span>")
+	if(signs.len)
+		for(var/H in signs)
+			qdel(H)
+		to_chat(user, "<span class='notice'>You clear all active holograms.</span>")
+
+//*******************************************
+//SEC HOLOBARRIER ITEM AND UPGRADE - ENDS
+//*******************************************
+
+//*******************************************
+//SEC INTEGRATED E-BOLA (lol) LAUNCHER ITEM AND UPGRADE - BEGINS
+//*******************************************
+
+/obj/item/gun/energy/e_gun/e_bola/cyborg
+	name = "\improper Integrated E-BOLA Launcher"
+	desc = "An integrated e-bola launcher that draws from a cyborg's power cell."
+	icon_state = "dragnet"
+	can_charge = FALSE
+	use_cyborg_cell = TRUE
+	charge_delay = 8
+	ammo_type = list(/obj/item/ammo_casing/energy/bola)
+
+/obj/item/borg/upgrade/e_bola
+	name = "cyborg energy bola launcher"
+	desc = "A module that permits firing energy bolas."
+	icon = 'icons/obj/guns/energy.dmi'
+	icon_state = "dragnet"
+	require_module = TRUE
+	module_type = list(/obj/item/robot_module/security)
+
+/obj/item/borg/upgrade/e_bola/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/gun/energy/e_gun/e_bola/cyborg/E = locate() in R.module.modules
+		if(E)
+			to_chat(user, "<span class='warning'>This unit already has a [E] installed!</span>")
+			return FALSE
+
+		E = new(R.module)
+		R.module.basic_modules += E
+		R.module.add_module(E, FALSE, TRUE)
+
+/obj/item/borg/upgrade/e_bola/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/gun/energy/e_gun/e_bola/cyborg/E = locate() in R.module.modules
+		if (E)
+			R.module.remove_module(E, TRUE)
+
+/obj/item/ammo_casing/energy/bola
+	projectile_type = /obj/projectile/energy/trap/cyborg
+	select_name = "bola"
+	e_cost = 400
+	harmful = FALSE
+
+//*******************************************
+//SEC INTEGRATED E-BOLA (lol) LAUNCHER ITEM AND UPGRADE - ENDS
+//*******************************************
+
+//*******************************************
+//SEC INTEGRATED ENERGY GUN ITEM AND UPGRADE - BEGINS
+//*******************************************
+
+/obj/item/gun/energy/e_gun/cyborg

```

Comment:
IMO, you should replace their disablers with energy guns that can only be fired on disabler mode, then, when they're emagged, allow them to switch to (and back to disabler mode from) lethal mode instead of giving them a separate laser gun. It'd reduce the amount of hotbar fiddling that emagged secborgs would have to do.

Reply:
I'd consider it; would need to adjust the cooler upgrade.